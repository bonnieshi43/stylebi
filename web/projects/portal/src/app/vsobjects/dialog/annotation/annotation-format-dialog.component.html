<!--
  ~ This file is part of StyleBI.
  ~ Copyright (C) 2024  InetSoft Technology
  ~
  ~ This program is free software: you can redistribute it and/or modify
  ~ it under the terms of the GNU Affero General Public License as published by
  ~ the Free Software Foundation, either version 3 of the License, or
  ~ (at your option) any later version.
  ~
  ~ This program is distributed in the hope that it will be useful,
  ~ but WITHOUT ANY WARRANTY; without even the implied warranty of
  ~ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
  ~ GNU Affero General Public License for more details.
  ~
  ~ You should have received a copy of the GNU Affero General Public License
  ~ along with this program.  If not, see <https://www.gnu.org/licenses/>.
  -->
<modal-header [title]="'_#(Format)'"
              [cshid]="'AnnotatingDashboard'"
              (onCancel)="cancel()">
</modal-header>
<div class="modal-body" enterSubmit (onEnter)="ok()" *ngIf="model">
  <form ngNoForm (submit)="$event.preventDefault()" class="container-fluid">
    <fieldset>
      <legend>_#(Box)</legend>
      <div class="form-row-float-label row">
        <div class="col pe-0">
          <div class="form-floating">
            <style-dropdown [(style)]="model.boxBorderStyle" [color]="model.boxBorderColor"
                            [noneAvailable]="false"></style-dropdown>
            <label>_#(Style)</label>
          </div>
        </div>
        <div class="col-auto ps-1">
          <color-editor [(color)]="model.boxBorderColor"></color-editor>
        </div>
      </div>
      <div class="form-row-float-label row">
        <div class="col">
          <div class="form-floating">
            <radius-dropdown [(radius)]="model.boxBorderRadius"
                             [form]="form.controls['roundCornerForm']"></radius-dropdown>
            <label>_#(Round Corner)</label>
          </div>
        </div>
        <div class="col">
          <color-editor [(color)]="model.boxFillColor" label="_#(Fill Color)"></color-editor>
        </div>
        <div class="col">
          <div class="form-floating">
            <alpha-dropdown [(alpha)]="model.boxAlpha"
                            (alphaInvalid)="changeAlphaWarning($event)"></alpha-dropdown>
            <label>_#(Alpha)</label>
          </div>
        </div>
      </div>
      <div *ngIf="alphaInvalid" class="col-12 alert alert-danger">
        _#(viewer.flash.format.invalidTransparencyError)
      </div>
    </fieldset>
    <fieldset *ngIf="showLine() && model.lineStyle">
      <legend>_#(Line)</legend>
      <div class="form-row-checkbox row">
        <div class="col checkbox">
          <div class="form-check">
            <input type="checkbox" class="form-check-input" id="visible"
                   [(ngModel)]="model.lineVisible">
            <label class="form-check-label" for="visible">
              _#(Visible)
            </label>
          </div>
        </div>
      </div>
      <div class="form-row-float-label row">
        <div class="col">
          <div class="form-floating">
            <style-dropdown [(style)]="model.lineStyle" [color]="model.lineColor"
                            [noneAvailable]="false">
            </style-dropdown>
            <label>_#(Line)</label>
          </div>
        </div>
        <div class="col">
          <div class="form-floating">
            <line-arrow-type-dropdown [(style)]="model.lineEnd" [color]="model.lineColor"></line-arrow-type-dropdown>
            <label>_#(End)</label>
          </div>
        </div>
        <div class="col-auto ps-0">
          <color-editor [(color)]="model.lineColor"></color-editor>
        </div>
      </div>
    </fieldset>
  </form>
</div>

<div class="modal-footer">
  <button type="button" class="btn btn-default reset-button" (click)="reset()">_#(Reset)</button>
  <button type="button" class="btn btn-primary" (click)="ok()" [disabled]="!form.valid ||
      alphaInvalid"
          #okButton>_#(OK)</button>
  <button type="button" class="btn btn-default" data-dismiss="modal" (click)="cancel()">_#(Cancel)</button>
</div>
