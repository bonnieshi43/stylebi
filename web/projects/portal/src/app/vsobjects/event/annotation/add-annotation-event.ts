/*
 * inetsoft-web - StyleBI is a business intelligence web application.
 * Copyright Â© 2024 InetSoft Technology (info@inetsoft.com)
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */
import { ViewsheetEvent } from "../../../common/viewsheet-client";

export class AddAnnotationEvent implements ViewsheetEvent {
   private content: string;
   private x: number;
   private y: number;
   private parent: string;
   private row: number = -1;
   private col: number = -1;
   private measureName: string;

   public setX(value: number) {
      this.x = value;
   }

   public setY(value: number) {
      this.y = value;
   }

   public setRow(row: number): void {
      this.row = row;
   }

   public setCol(col: number): void {
      this.col = col;
   }

   public setMeasureName(measureName: string): void {
      this.measureName = measureName;
   }

   constructor(content: string, x: number, y: number, parent?: string) {
      this.content = content;
      this.x = x;
      this.y = y;
      this.parent = parent;
   }
}
