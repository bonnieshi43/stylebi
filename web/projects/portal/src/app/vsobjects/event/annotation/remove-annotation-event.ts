/*
 * This file is part of StyleBI.
 * Copyright (C) 2024  InetSoft Technology
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */
import { ViewsheetEvent } from "../../../common/viewsheet-client";
import { VSObjectModel } from "../../model/vs-object-model";

export class RemoveAnnotationEvent implements ViewsheetEvent {
   public static readonly REMOVE_ANNOTATION_URI: string = "/events/annotation/remove-annotation";
   // The names of the annotations to remove
   private names: string[];

   private constructor(names: string[]) {
      this.names = names;
   }

   /**
    * Creates a RemoveAnnotationEvent that contains the names of the annotations to be removed
    *
    * @param vsObjects          the objects to check for annotations
    * @param selectedAssemblies the selected objects
    */
   public static create(vsObjects: VSObjectModel[],
                        selectedAssemblies: number[]): RemoveAnnotationEvent
   {
      if(selectedAssemblies) {
         const names = selectedAssemblies
            .map(index => vsObjects[index])
            .reduce((selected, current) => {
               if(current.objectType === "VSAnnotation") {
                  return selected.concat(current.absoluteName);
               }
               else if(current.selectedAnnotations) {
                  return selected.concat(current.selectedAnnotations);
               }
               else {
                  return selected;
               }
            }, []);

         if(names.length > 0) {
            return new RemoveAnnotationEvent(names);
         }
      }

      return null;
   }
}
