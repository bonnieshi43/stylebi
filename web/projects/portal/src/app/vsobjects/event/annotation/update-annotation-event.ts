/*
 * inetsoft-web - StyleBI is a business intelligence web application.
 * Copyright Â© 2024 InetSoft Technology (info@inetsoft.com)
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */
import { ViewsheetEvent } from "../../../common/viewsheet-client";
import { Rectangle } from "../../../common/data/rectangle";

export class UpdateAnnotationEvent implements ViewsheetEvent {
   // Name of annotation to update
   private name: string;

   // Change in top/left/width/height of annotation
   private newBounds: Rectangle;

   // The new content for the annotation
   private content: string;

   constructor(name: string, newBounds: Rectangle, content?: string) {
      this.name = name;
      this.newBounds = newBounds;
      this.content = content;
   }
}