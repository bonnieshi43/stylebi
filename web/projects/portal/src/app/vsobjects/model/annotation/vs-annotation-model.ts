/*
 * inetsoft-web - StyleBI is a business intelligence web application.
 * Copyright Â© 2024 InetSoft Technology (info@inetsoft.com)
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affrero General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */
import { VSObjectModel } from "../vs-object-model";
import { VSLineModel } from "../vs-line-model";
import { VSRectangleModel } from "../vs-rectangle-model";
import { HtmlContentModel } from "../html-content-model";
import { Point } from "../../../common/data/point";

export interface VSAnnotationModel extends VSObjectModel {
   annotationRectangleModel: VSRectangleModel;
   annotationLineModel: VSLineModel;
   contentModel: HtmlContentModel;

   // row and col in table used to identify which cell contains the annotation
   row: number;
   col: number;

   // true if annotation status is hidden on the viewsheet
   hidden: boolean;

   // {viewsheet, assembly, data}
   annotationType: number;

   // data annotation may need runtime position information
   cellOffset?: Point;
}
