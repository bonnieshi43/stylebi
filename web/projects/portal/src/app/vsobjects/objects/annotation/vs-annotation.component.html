<!--
  ~ This file is part of StyleBI.
  ~ Copyright (C) 2024  InetSoft Technology
  ~
  ~ This program is free software: you can redistribute it and/or modify
  ~ it under the terms of the GNU Affero General Public License as published by
  ~ the Free Software Foundation, either version 3 of the License, or
  ~ (at your option) any later version.
  ~
  ~ This program is distributed in the hope that it will be useful,
  ~ but WITHOUT ANY WARRANTY; without even the implied warranty of
  ~ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
  ~ GNU Affero General Public License for more details.
  ~
  ~ You should have received a copy of the GNU Affero General Public License
  ~ along with this program.  If not, see <https://www.gnu.org/licenses/>.
  -->
<ng-container *ngIf="model && model.visible && model.annotationRectangleModel">
  <ng-container *ngIf="!model.hidden; else annotationPlaceholder">
    <ng-container *ngIf="isLineInContainer()">
      <ng-container *ngIf="model.annotationLineModel">
        <vs-line [model]="model.annotationLineModel">
        </vs-line>
        <div *ngIf="isAssemblyAnnotation()"
             class="vs-annotation__line-endpoint hover-bg-white-border"
             [style.top.px]="model.annotationLineModel.endTop + model.annotationLineModel.objectFormat.top"
             [style.left.px]="model.annotationLineModel.endLeft + model.annotationLineModel.objectFormat.left"
             [style.z-index]="model.annotationLineModel.objectFormat.zIndex"
             wInteractable
             [interactableDraggable]="true"
             [draggableRestriction]="restrictTo"
             [composited]="false"
             (onDraggableMove)="onLineEndDragMove($event)"
             (onDraggableEnd)="onLineEndDragEnd($event)">
        </div>
      </ng-container>
      <div class="vs-annotation__rectangle vs-object"
           [style.top.px]="model.annotationRectangleModel.objectFormat.top"
           [style.left.px]="model.annotationRectangleModel.objectFormat.left"
           [style.width.px]="model.annotationRectangleModel.objectFormat.width"
           [style.height.px]="model.annotationRectangleModel.objectFormat.height"
           [style.z-index]="model.annotationRectangleModel.objectFormat.zIndex"
           [class.vs-annotation__rectangle--selected]="selected"
           [class.active]="selected"
           draggable="true"
           wInteractable
           [resizableMargin]="8"
           [interactableDraggable]="selected"
           [interactableResizable]="selected"
           [resizableTopEdge]="'.top'"
           [resizableBottomEdge]="'.bottom'"
           [resizableLeftEdge]="'.left'"
           [resizableRightEdge]="'.right'"
           [interactableDraggableGroup]="'vs-annotation-rectangle-drag-group'"
           outOfZone
           (onMouseenter)="checkOverflow()"
           (onDraggableMove)="onRectangleDragMove($event)"
           (onDraggableEnd)="updateAnnotation($event)"
           (onResizableMove)="onRectangleResizeMove($event)"
           (onResizableEnd)="updateAnnotation($event)">
        <img class="vs-annotation__rectangle-image" *ngIf="vsInfo?.linkUri"
             [title]=""
             [src]="getSrc()"/>
        <div class="vs-annotation__rectangle-content-padding-box"
             [style.padding.px]="contentPadding"
             [class.vs-annotation__rectangle-content--selected]="selected"
             [wTooltip]="isOverflowing ? content : ''">
          <div class="vs-annotation__rectangle-content">
            <p class="vs-annotation__rectangle-content-text" #annotationContent
               [innerHtml]="content">
            </p>
          </div>
        </div>
        <ng-container *ngIf="selected">
          <div class="container-gutter top"></div>
          <div class="container-gutter bottom"></div>
          <div class="container-gutter left"></div>
          <div class="container-gutter right"></div>
          <div class="container-gutter top left"></div>
          <div class="container-gutter top right"></div>
          <div class="container-gutter bottom left"></div>
          <div class="container-gutter bottom right"></div>
        </ng-container>
      </div>
    </ng-container>
  </ng-container>
  <ng-template #annotationPlaceholder>
    <div *ngIf="isViewsheetAnnotation()"
         class="vs-annotation__hidden-icon--viewsheet" aria-hidden="true"
         [style.top.px]="model.annotationRectangleModel.objectFormat.top"
         [style.left.px]="model.annotationRectangleModel.objectFormat.left">
      <button type="button" class="btn-sm annotition-button" aria-label="Toggle Annotations"
              (click)="toggleAnnotationStatus()"
              [wTooltip]="content">
        <span class="annotation-icon" aria-hidden="true"></span>
      </button>
    </div>
  </ng-template>
</ng-container>
