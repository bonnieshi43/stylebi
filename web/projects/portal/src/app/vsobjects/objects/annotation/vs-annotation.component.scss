/**
 * inetsoft-web - StyleBI is a business intelligence web application.
 * Copyright Â© 2024 InetSoft Technology (info@inetsoft.com)
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */
$line-handle-height: 10px;
$line-handle-width: 10px;

.vs-annotation:active > vs-line {
  visibility: hidden;
}

.vs-annotation__line-endpoint {
  position: absolute;
  width: $line-handle-width;
  height: $line-handle-height;
  margin-top: -$line-handle-height * 0.5;
  margin-left: -$line-handle-width * 0.5;
  background: transparent; // left as not themeable
  border: none;
}

.vs-annotation__rectangle {
  position: absolute;
  overflow: hidden;
}

.vs-annotation__rectangle--selected {
  border: 2px dotted darkgray;
}

.vs-annotation__rectangle-content-padding-box {
  position: absolute;
  overflow: hidden;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;

  .vs-annotation__rectangle-content {
    width: 100%;
    height: 100%;
    overflow: auto;
    color: black;
  }

  .vs-annotation__rectangle-content-text {
    display: flex;
    flex-direction: column;
    margin: 0;

    ::ng-deep p {
      overflow-wrap: break-word;
      white-space: pre-wrap;
      margin: 0;
      -webkit-margin-before: 0;
      -webkit-margin-after: 0;
    }
  }
}

.vs-annotation__rectangle-content--selected {
  //resize: both;
}

.vs-annotation__rectangle-image {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
}

.vs-annotation__hidden-icon--viewsheet {
  position: absolute;
}

.annotition-button {
  border: 2px outset  #cccccc;
}
