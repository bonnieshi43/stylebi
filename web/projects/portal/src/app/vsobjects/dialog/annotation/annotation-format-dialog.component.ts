/*
 * inetsoft-web - StyleBI is a business intelligence web application.
 * Copyright Â© 2024 InetSoft Technology (info@inetsoft.com)
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */
import { HttpClient } from "@angular/common/http";
import { Component, EventEmitter, Input, OnInit, Output } from "@angular/core";
import { UntypedFormGroup } from "@angular/forms";
import { Annotation } from "../../../common/data/annotation";
import { AnnotationFormatDialogModel } from "./annotation-format-dialog-model";

const INIT_VIEWSHEET_FORMAT_URI = "../annotation/resetViewsheetFormat";
const INIT_ASSEMBLY_FORMAT_URI = "../annotation/resetAssemblyFormat";
const INIT_DATA_FORMAT_URI = "../annotation/resetDataFormat";

@Component({
   selector: "annotation-format-dialog",
   templateUrl: "annotation-format-dialog.component.html",
})
export class AnnotationFormatDialog implements OnInit {
   @Input() model: AnnotationFormatDialogModel;
   @Input() annotationType: Annotation.Type;
   @Input() forChart: boolean = false;
   @Output() onCommit = new EventEmitter<AnnotationFormatDialogModel>();
   @Output() onCancel = new EventEmitter<string>();
   form: UntypedFormGroup;
   alphaInvalid: boolean = false;

   ngOnInit(): void {
      this.form = new UntypedFormGroup({
         roundCornerForm: new UntypedFormGroup({})
      });
   }

   constructor(private http: HttpClient) {
   }

   public showLine() {
      return !Annotation.isViewsheetAnnotation(this.annotationType);
   }

   /**
    * Reset the dialog to the initialized state.
    */
   async reset() {
      if(this.forChart) {
         this.model = await this.http.get<AnnotationFormatDialogModel>(INIT_ASSEMBLY_FORMAT_URI).toPromise();
         return;
      }

      const uris: [string, (t) => boolean][] = [
         [INIT_VIEWSHEET_FORMAT_URI, Annotation.isViewsheetAnnotation],
         [INIT_ASSEMBLY_FORMAT_URI, Annotation.isAssemblyAnnotation],
         [INIT_DATA_FORMAT_URI, Annotation.isDataAnnotation]
      ];

      const [matchedUri] = uris.find(([uri, predicate]) => predicate(this.annotationType));
      this.model = await this.http.get<AnnotationFormatDialogModel>(matchedUri).toPromise();
   }

   ok() {
      this.onCommit.emit(this.model);
   }

   cancel() {
      this.onCancel.emit("cancel");
   }

   changeAlphaWarning(event) {
      this.alphaInvalid = event;
   }
}
