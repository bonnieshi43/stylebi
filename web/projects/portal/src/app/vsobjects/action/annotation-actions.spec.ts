/*
 * inetsoft-web - StyleBI is a business intelligence web application.
 * Copyright Â© 2024 InetSoft Technology (info@inetsoft.com)
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */
import { TestUtils } from "../../common/test/test-utils";
import { ViewerContextProviderFactory } from "../context-provider.service";
import { AnnotationActions } from "./annotation-actions";
import { VSAnnotationModel } from "../model/annotation/vs-annotation-model";

describe("AnnotationActions", () => {
   const createModel = () => TestUtils.createMockVSAnnotationModel("Annotation1");

   it("check menu actions is created", () => {
      const expectedMenu = (srcModel: VSAnnotationModel) => [
         [
            { id: "annotation edit" + srcModel.absoluteName, visible: true },
            { id: "annotation format" + srcModel.absoluteName, visible: true },
            { id: "annotation remove" + srcModel.absoluteName, visible: true }
         ]
      ];

      //check status in viewer
      const model = createModel();
      const actions1 = new AnnotationActions(model, ViewerContextProviderFactory(false));
      const menuActions1 = actions1.menuActions;

      expect(menuActions1).toMatchSnapshot();

      //check status in preview
      const model2 = createModel();
      const actions2 = new AnnotationActions(model2, ViewerContextProviderFactory(true));
      const menuActions2 = actions2.menuActions;

      expect(menuActions2).toMatchSnapshot();
   });
});