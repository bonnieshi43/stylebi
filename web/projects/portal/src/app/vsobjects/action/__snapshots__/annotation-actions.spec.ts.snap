// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnnotationActions check menu actions is created 1`] = `
Array [
  AssemblyActionGroup {
    "actions": Array [
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
    ],
    "icon": [Function],
    "label": [Function],
  },
]
`;

exports[`AnnotationActions check menu actions is created 2`] = `
Array [
  AssemblyActionGroup {
    "actions": Array [
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
    ],
    "icon": [Function],
    "label": [Function],
  },
]
`;
